src/movepick.cpp

```diff
@@ -137,6 +137,17 @@ void MovePicker::score() {
                           :                                         !(to_sq(m) & threatenedByPawn)  ? 15000
                           :                                                                           0)
                           :                                                                           0)
+                   -     (!(threatenedPieces & from_sq(m)) ?
+                           (type_of(pos.moved_piece(m)) == QUEEN ?  
+                                (bool(to_sq(m) & threatenedByRook)  * 50000 
+                               + bool(to_sq(m) & threatenedByMinor) * 10000 
+                               + bool(to_sq(m) & threatenedByPawn)  * 20000)
+                          : type_of(pos.moved_piece(m)) == ROOK  ?
+                                (bool(to_sq(m) & threatenedByMinor) * 25000 
+                               + bool(to_sq(m) & threatenedByPawn)  * 10000)
+                          : type_of(pos.moved_piece(m)) != PAWN  && (to_sq(m) & threatenedByPawn)  ? 15000
+                          :                                                                           0)
+                          :                                                                           0)
```

Comment:
There are some small whitespace/indentation issue:

```suggestion
                   -     (!(threatenedPieces & from_sq(m)) ?
                           (type_of(pos.moved_piece(m)) == QUEEN ?
                                (bool(to_sq(m) & threatenedByRook)  * 50000
                               + bool(to_sq(m) & threatenedByMinor) * 10000
                               + bool(to_sq(m) & threatenedByPawn)  * 20000)
                          : type_of(pos.moved_piece(m)) == ROOK  ?
                                (bool(to_sq(m) & threatenedByMinor) * 25000
                               + bool(to_sq(m) & threatenedByPawn)  * 10000)
                          : type_of(pos.moved_piece(m)) != PAWN  && (to_sq(m) & threatenedByPawn) ? 15000
                          :                                                                         0)
                          :                                                                         0)
```